### PR TITLE
Added special char validation. Used it and trimValidator to name field on create contact page

### PR DESCRIPTION
**Pull request by VKislov**
specCharValidator:
Screenshot 2022-10-13 at 13 44 59
trimValidator: 
Screenshot 2022-10-13 at 13 47 09

### CODE REVIEW

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 1]
```diff
@@ -292,7 +298,15 @@ export const CreateContactForm: React.FC = () => {
                                                                 onClick={handleSave}
                                                                 type='sberPrimary'
                                                                 loading={isLoading}
-                                                                disabled={!property || !unitName || !phone || !name || !!propertyMismatchError || hasContactDuplicate}
+                                                                disabled={
+                                                                    !property ||
+                                                                    !unitName ||
+                                                                    !phone ||
```
Comment (sitozzz) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:304]: Please extract this condition to variable with clear name

File: apps/condo/domains/common/hooks/useValidations.ts [hunk 2]
```diff
@@ -66,6 +67,18 @@ export const useValidations: UseValidations = (settings = {}) => {
         },
     }
 
+    const specCharValidator: Rule = {
+        validator: (_, value) => {
+            if (value) {
+                const invalidCharRegex = /[~!@#±$"№%^&*()_+=[\]{}|\\;’:”,./<>?£]/
```
Comment (sitozzz) [inline apps/condo/domains/common/hooks/useValidations.ts:73]: Can this regex be reused in the future? May be move it to constant?

**Review by akarjakin — APPROVED** [review]

File: apps/condo/domains/common/hooks/useValidations.ts (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/domains/common/hooks/useValidations.ts:73]: Maybe, 
```
/[^A-ZА-Я0-9\s\-]/i
```

File: apps/condo/domains/contact/components/ErrorsContainer.tsx [hunk 3]
```diff
@@ -10,30 +10,46 @@ interface IErrorsContainerProps {
...
     if (!phone) messageLabels.push(`"${PhoneLabel}"`)
 
     const requiredErrorMessage = !isEmpty(messageLabels) && ErrorsContainerTitle.concat(' ', messageLabels.join(', '))
     const contactDuplicateError = hasContactDuplicate ? contactDuplicateMessage : undefined
+    const nameSpecCharError = hasNameSpecCharError ? FullNameInvalidCharMessage : undefined
+
     return (
         disabledUserInteractions && (
             <ErrorsWrapper>
                 <div>{contactDuplicateError}</div>
```
[the remark this answers is not in the document]
Comment (VKislov) [reply apps/condo/domains/contact/components/ErrorsContainer.tsx:51]: I tried, the spaces between the lines are too large

File: apps/condo/domains/common/hooks/useValidations.ts (hunk 2 above)
Comment (VKislov) [reply apps/condo/domains/common/hooks/useValidations.ts:73]: Yes we can, I'll take it out in a constant

File: apps/condo/domains/contact/components/CreateContactForm.tsx [hunk 4]
```diff
@@ -281,8 +285,10 @@ export const CreateContactForm: React.FC = () => {
                                             const { phone, property, unitName, name } = getFieldsValue(['phone', 'property', 'unitName', 'name'])
                                             const propertyMismatchError = getFieldError('property').find((error)=>error.includes(AddressNotSelected))
                                             const hasDuplicateError = Boolean(getFieldError('_NON_FIELD_ERROR_').find((error => error.includes(ContactDuplicateError))))
+                                            const hasNameSpecCharError = Boolean(getFieldError('name').find((error => error.includes(FullNameInvalidCharMessage))))
+                                            const hasNameTrimValidateError = Boolean(getFieldError('name').find((error => error.includes(FullNameRequiredMessage))))
                                             const hasContactDuplicate = isFieldsChanged ? false : hasDuplicateError
-
+                                            const disabledCondition = !property || !unitName || !phone || hasNameTrimValidateError || !!propertyMismatchError || hasContactDuplicate || hasNameSpecCharError
```
Comment (sitozzz) [inline apps/condo/domains/contact/components/CreateContactForm.tsx:291]:
```suggestion
                                            const isDisabled = !property || !unitName || !phone || hasNameTrimValidateError || !!propertyMismatchError || hasContactDuplicate || hasNameSpecCharError
```

**Pull request merged**